### DISCUSSION TITLE
How to deploy Next.js 9.5 with Dockerfile on Traefik

### DISCUSSION THREAD

**Discussion by fatihyildizhan** [vercel/next.js#16411] [Help] [answered]
Hi, 

I have a simple next.js project. I am using `next/head`. I want to publish it on Traefik as a Docker container. I am getting Bad Gateway error. How can I fix this error?

Dockerfile
```
FROM node:12.18.3-alpine3.12

COPY . /app
WORKDIR /app

RUN npm install

RUN npm run build

EXPOSE 3000

CMD [ "npm", "run", "start" ]
```

.dockerignore
```
.next/
node_modules/
.git
.gitignore
Dockerfile
```

Server port is 80 on Traefik's docker-compose.yml
`- "traefik.http.services.simple-nextjs.loadbalancer.server.port=80"`

**Comment by c0b41** [comment, marked as answer]
change port to `3000`

**Reply by fatihyildizhan** [reply]
I changed server port to 3000 on container's docker-compose.yml

"traefik.http.services.simple-nextjs.loadbalancer.server.port=3000"
It worked,
Thank you very much

**Comment by cristiand391** [comment]


**Reply by c0b41** [reply]
you can't run nodejs 80 port, he need to change loadbalancer port
